### DISCUSSION TITLE
Auto convert % and format to f-string

### DISCUSSION THREAD

**Discussion by asukaminato0721** [microsoft/pylance-release#2639] [Enhancement]
There are three types of formatting a string. Is it possible to add a yellow bulb to suggest converting the last two to f-string automatically?

```py
a = "str"
f"{a}"
"%s" % (a)
"{}".format(a)
```

**Comment by judej** [comment]
Thanks for the suggestion @wuyudi, will add this to our enhancement list.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by 27Bslash6** [comment]
Or vice-versa, a way to convert an f-string to one of the latter two, as I repeatedly find myself adding new variables to fstrings and then needing to rewrite to avoid hitting a line-length limit.

**Comment by asukaminato0721** [comment]
I find this can do auto-convert, and many more other things.

https://sourcery.ai/

**Comment by brettcannon** [comment]
There's also the case of simply forgetting the `f` prefix: `"{a}"`.

**Reply by jxramos** [reply]
while developing in this space the inverse would be useful to detect as well, perhaps adding a squiggly underline when a string has an `f` prefix for a string whose body is just a plain string literal with no evaluation tokens within it, eg
* `f"this is just a string"` <--------- give this a red or yellow squiggly
* `f"this is a string with an evaluation token {3}"` <------ no squiggly for this case

**Comment by asukaminato0721** [comment]
oh, this solves it https://github.com/ikamensh/flynt
